modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/podcast/UserPlaylistUpdate.kt

```diff
@@ -0,0 +1,12 @@
+package au.com.shiftyjelly.pocketcasts.repositories.podcast
+
+data class UserPlaylistUpdate(
+    val properties: List<PlaylistProperty>,
+    val source: PlaylistUpdateSource
+) {
```

Comment:
I like the approach you took with passing `userPlaylistUpdate` to `PlaylistManagerImpl.update(...)` and understand [your earlier comment](PlaylistManagerImpl) better now. As you mentioned, this approach has a type-checking advantage.

For filters, it is working very well. I know, we need to resolve unknown sources for playback events and maybe this approach can help. I'll discuss it in a separate PR once I have a better idea about the complexity.